group/assignment6/RearrangeCars.cpp

```diff
@@ -139,3 +139,34 @@ void RearrangeCars::generateSequenceOfMovesWithConstraints(const vector<int> &en
 			break;
 	}
 }
+
+void RearrangeCars::bruteForceAllSequence(vector<int>& currentStates, const vector<int>& endStates, vector<int>& carPosition, int emptySlotId, vector<Move>& sequenceOfMoves, vector<vector<Move>>& sequencesOfMoves, set<vector<int>>& usedPositions) {
+	int numOfCarsOnDisiredPosition = numberOfCarsOnDesiredPosition(carPosition, endStates);
+	if (numOfCarsOnDisiredPosition == carPosition.size()) {
+		sequencesOfMoves.push_back(sequenceOfMoves);
+		return;
+	}
+	for (int currCarSlotId = 0; currCarSlotId < carPosition.size(); currCarSlotId++) {
+		if (currCarSlotId != emptySlotId) {
+			moveCar(currentStates, carPosition, currCarSlotId, emptySlotId);
+			if (usedPositions.find(carPosition) == usedPositions.end()) {
+				usedPositions.insert(carPosition);
+				sequenceOfMoves.push_back(Move(currentStates[emptySlotId], make_pair(currCarSlotId, emptySlotId)));
+				bruteForceAllSequence(currentStates, endStates, carPosition, currCarSlotId, sequenceOfMoves, sequencesOfMoves, usedPositions);
+				sequenceOfMoves.pop_back();

```

Comment:
This section was confusing to me - I understand that we swapped the car in _currCarSlotId_ with the "car" in _emptySlotId_, but it's really hard to understand that the values in _currentStates_ changed when you first read it. Maybe we can also swap the values in _currCarSlotId_ and _emptySlotId_ after we push the move to the sequence of moves, or at least add a comment that explains what's happening here. What do you think?